crates/sui-light-client/src/main.rs

```diff
@@ -202,73 +263,58 @@ async fn sync_checkpoint_list_to_latest(config: &Config) -> anyhow::Result<()> {
     // Download the latest in list checkpoint
     let summary = download_checkpoint_summary(config, *latest_in_list).await?;
     let mut last_epoch = summary.epoch();
-    let mut last_checkpoint_seq = summary.sequence_number;
 
     // Download the very latest checkpoint
-    let client = Client::new(config.rest_url());
-    let latest = client.get_latest_checkpoint().await?;
+    let client = SuiClientBuilder::default()
+        .build(config.full_node_url.as_str())
+        .await
+        .expect("Cannot connect to full node");
+
+    let latest_seq = client
+        .read_api()
+        .get_latest_checkpoint_sequence_number()
+        .await?;
+    let latest = download_checkpoint_summary(config, latest_seq).await?;
 
     // Binary search to find missing checkpoints
```

Comment:
no longer a binary search?

Reply:
Good catch.

Also you did catch a bug when the events list is None (ie. no events).